activemodel/lib/active_model/attribute_assignment.rb

```diff
@@ -26,13 +26,12 @@ module AttributeAssignment
     #   cat.name # => 'Gorby'
     #   cat.status # => 'sleeping'
     def assign_attributes(new_attributes)
-      if !new_attributes.respond_to?(:stringify_keys)
+      unless new_attributes.respond_to?(:stringify_keys)
```

Comment:
We no longer call `stringify_keys`, so this guard should be updated. We could check for the `each` method instead, since that's what's eventually called on the object, but allowing any enumerable seems too broad - passing an array of attribute names would unexpectedly assign each of them as nil, for example.

Perhaps we should update both this guard and the `_assign_attributes` methods to use `each_pair` instead?

Reply:
Do we actually need the guard at all? Usually, iirc, we let usage examples in docs help guide users and skip rescuing type errors.

Reply:
The guard was added to address https://github.com/rails/rails/issues/9858, where a typo in a parameter name would be silently ignored. As of https://github.com/rails/rails/pull/28642 passing nil would cause a `NoMethodError` if the guard was removed, so I'm 👍 to ✂️ it.

My concern about unexpectedly allowing arrays would still stand, though.

Reply:
I like the idea of checking for `each_pair`, if we want to avoid having it called with an array.